node/actors/control_api.go

```diff
@@ -1,26 +1,197 @@
 package actors
 
 import (
+	"errors"
+	"fmt"
+	"log/slog"
+
 	"ergo.services/ergo/act"
 	"ergo.services/ergo/gen"
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
 )
 
-func createControlAPIServer() gen.ProcessBehavior {
-	return &controlAPIServer{}
+const APIPrefix = "$NEX"
+
+func createControlAPI() gen.ProcessBehavior {
+	return &controlAPI{
+		subsz: make([]*nats.Subscription, 0),
+	}
 }
 
-type controlAPIServer struct {
+type controlAPI struct {
 	act.Actor
+
+	publicKey  string
+	publicXKey string
+	subsz      []*nats.Subscription
 }
 
-func (api *controlAPIServer) Init(args ...any) error {
-	api.Log().Info("Control API Server started")
+func (api *controlAPI) Init(args ...any) error {
+	if len(args) != 2 {
+		err := errors.New("NATS connection and public key are required")
+		api.Log().Error("Failed to start control API", slog.String("error", err.Error()))
+		return err
+	}
+
+	if _, ok := args[0].(*nats.Conn); !ok {
+		err := errors.New("args[0] must be a valid NATS connection")
+		api.Log().Error("Failed to start control API", slog.String("error", err.Error()))
+		return err
+	}
+
+	if publicKey, ok := args[1].(string); ok {
+		api.publicKey = publicKey
+	}
+
+	err := api.initPublicXKey()
+	if err != nil {
+		return err
+	}
+
+	err = api.subscribe(args[0].(*nats.Conn))
+	if err != nil {
+		_ = api.shutdown()
+		return err
+	}
+
+	api.Log().Info("Control API started",
+		slog.String("public_key", api.publicKey),
+		slog.String("public_xkey", api.publicXKey),
+	)
 
 	return nil
 }
 
 // HandleInspect invoked on the request made with gen.Process.Inspect(...)
-func (api *controlAPIServer) HandleInspect(from gen.PID, item ...string) map[string]string {
+func (api *controlAPI) HandleInspect(from gen.PID, item ...string) map[string]string {
 	api.Log().Info("control API server got inspect request from %s", from)
 	return nil
 }
+
+func (api *controlAPI) initPublicXKey() error {
+	kp, err := nkeys.CreateCurveKeys()
+	if err != nil {
+		api.Log().Error("Failed to create x509 curve key", slog.Any("error", err))
```

Comment:
xkeys are not x.509

Reply:
Good catch... this is an artifact from copying code over from main.